deployment/aws/terraform/main.tf

```diff
@@ -1,4 +1,10 @@
 module "aws" {
-  source              = "../../../iac/terraform/aws"
-  environment         = var.aws_environment
+  source         = "../../../iac/terraform/aws"
+  environment    = var.aws_environment
+  hosted_zone_id = data.aws_route53_zone.route53_zone.zone_id
 }
+
+data "aws_route53_zone" "route53_zone" {
```

Comment:
If roite53 zone is not necessary in deployment directory, pass domain name to iac directory and move this code there